src/queues.c

```diff
@@ -343,68 +375,32 @@ void queues_history_push_all(void)
 }
 
 /* see queues.h */
-void queues_check_timeouts(struct dunst_status status)
+void queues_update(struct dunst_status status)
```

Comment:
These are 2 huge and complicated functions, let's break them down rather than merge them.

Reply:
I don't see any benefit of splitting this method.

There are 4 parts, which are required to be executed in exactly this order and the logics involved is nowhere else necessary.

Additonally, the complex testing logic is already splitted out into `queues_notification_is_(ready|finished)`.